cyf-ecommerce-api/server.js

```diff
@@ -0,0 +1,364 @@
+      FROM product_availability 
+      INNER JOIN products ON products.id = product_availability.prod_id 
+      INNER JOIN suppliers ON suppliers.id = product_availability.supp_id 
+      WHERE product_availability.prod_id = $1 AND product_availability.supp_id = $2`,
+      [productId, supplierId]
+    )
+    .then((result) => {
+      if (result.rows.length == 0) {
+        return res.status(400).json({
+          result: "failure",
+          msg: `Id not found`,
+        });
+      } else {
+        const query = `SELECT prod_id, supp_id FROM product_availability WHERE prod_id = $1 AND supp_id = $2`;
+        pool.query(query, [productId, supplierId]).then((result) => {
+          if (result.rows.length > 0) {
+            return res.status(400).json({
+              result: "failure",
+              msg: "This product availability already exists",
+            });
+          } else {
+            const query =
+              "INSERT INTO product_availability (prod_id, supp_id, unit_price) VALUES($1, $2, $3)";
+            pool
+              .query(query, [productId, supplierId, price])
+              .then(() =>
+                res.json({
+                  msg: "A new product availability was added successful",
+                })
+              )
+              .catch((error) => {
+                res.status(400).json(error);
+              });
+          }
+        });
+      }
+    })
+    .catch((error) => res.status(400).json(error));
+});
+
+//- Add a new POST endpoint `/customers/:customerId/orders` to create a new order (including an order date, and an order reference) for a customer. Check that the customerId corresponds to an existing customer or return an error.
+app.post("/customers/:customerId/orders", (req, res) => {
+  const orderDate = req.body.order_date;
+  const orderReference = req.body.order_reference;
+  const customerId = Number(req.params.customerId);
+
+  if (!orderDate) {
+    return res.status(400).json({
+      result: "failure",
+      msg: "An orderDate value is required.",
+    });
+  } else if (!orderReference) {
+    return res.status(400).json({
+      result: "failure",
+      msg: "An orderReference value is required.",
+    });
+  }
+
+  pool
+    .query(
+      `SELECT * FROM customers INNER JOIN orders ON orders.customer_id = customers.id WHERE customers.id = $1`,
+      [customerId]
+    )
+    .then((result) => {
+      if (result.rows.length == 0) {
+        return res.status(400).json({
+          result: "failure",
+          msg: `Invalid customerId value.`,
+        });
+      } else {
+        const query = `INSERT INTO orders (order_date, order_reference, customer_id) VALUES($1, $2, $3)`;
+        pool
+          .query(query, [orderDate, orderReference, customerId])
+          .then(() =>
+            res.json({
+              msg: "Order has been successful created.",
+            })
+          )
+          .catch((error) => {
```

Comment:
we have to double check this but I m sure from somewhere one is enough at the end kind of catch-all